src/Bedrock.Framework/Protocols/ProtocolReadResult.cs

```diff
@@ -9,8 +13,27 @@ public ProtocolReadResult(TMessage message, bool isCanceled, bool isCompleted)
             IsCompleted = isCompleted;
         }
 
+        /// <summary>
+        /// The read message.
+        /// </summary>
         public TMessage Message { get; }
+
+        /// <summary>
+        /// Whether the reading operation was cancelled (true) or not (false).
+        /// </summary>
         public bool IsCanceled { get; }
+
+        /// <summary>
+        /// Whether the reading operation was completed (true) or not (false). 
+        /// </summary>
         public bool IsCompleted { get; }
+
+        /// <summary>
+        /// Whether the reading operation was faulty (true) or not (false).
+        /// </summary>
+        /// <remarks>
+        /// This property will be true if <see cref="Message"/> was null. 
+        /// </remarks>
+        public bool IsFaulted => Message is null;
```

Comment:
Not sure about this. Why do we need it?

Reply:
I can benefit from this property by directly using it from the result itself instead of having to access the result's message. It's just a small QOL improvement. It is not needed.

Reply:
Can you revert it then. New public API shouldn't be added in PRs like this.

Reply:
Of course. 😃